layers/fuzz_layer.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Catena cyber All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+
+	"github.com/google/gopacket"
+)
+
+var fuzzOpts = gopacket.DecodeOptions{
+	Lazy:                     false,
+	NoCopy:                   true,
+	SkipDecodeRecovery:       true,
+	DecodeStreamsAsDatagrams: true,
+}
+
+func FuzzLayer(data []byte) int {
+	if len(data) < 2 {
+		return 1
+	}
+	startLayer := binary.BigEndian.Uint16(data[:2])
```

Comment:
could you add a comment on what you're doing here?  My impression is that you're allowing the fuzzer to chose its own top-level layer by using the first 2 bytes to get it?

Reply:
Ok.
Your impression is correct.
This is kind of a speed optimization, so as not to parse Ethernet/IP/TCP over again to finally get to the application layer.
Plus there may be protocols that I do not know that are not reachable if we start from Ethernet...